py/torch_tensorrt/dynamo/lowering/_decompositions.py

```diff
@@ -162,6 +162,35 @@ def var_decomposition(
     return variance
 
 
+@register_torch_trt_decomposition(
+    torch.ops.aten.select_scatter.default, registry=TORCH_TRT_DECOMPOSITIONS
+)
+def select_scatter_decomposition(
+    input_tensor: torch.Tensor,
+    src_tensor: torch.Tensor,
+    dim: int,
+    index: int,
+) -> torch.Tensor:
+    # input_tensor.shape[dim] = torch.le(index, input_tensor.shape[dim])
+    # check if the dim is less than shape
+    if input_tensor.shape[dim] < index:
+        raise AssertionError("The index should not be greater than dim")
+
+    # expanding the src_tensor to have the same dimension as input_tensor
+    # check if the dimension of the src tensor is same as slice tensor
+    select_tensor = torch.select(input_tensor, dim, index)
+
+    if select_tensor.shape != src_tensor.shape:
+        raise AssertionError(
+            "The slice tensor shape should be equal to the src tensor shape"
+        )
```

Comment:
Are the `AssertionError` cases invalid in Torch, or just cases we can't support? Having `AssertionError`s in lowering passes can cause models to inexplicably fail for users, so it is not preferable.

If these are invalid cases in Torch itself, then we do not need these assertions. If they are not supported by TRT, then we can instead return the original op and not lower.

Reply:
Yes it is invalid test case in torch itself. So I think the right thing would be to do away with the assertion.
When you say return the original op, that would mean in those cases, we just do
```
if(condition == True):
     return <unlowered_original_op>
```

Reply:
Ok thanks for that clarification. I meant that if it would be invalid in Torch, you can assume it will not be the case that the condition would ever be encountered - otherwise the model should have failed earlier. Specifically, if `select_tensor.shape != src_tensor.shape` is a _requirement_ of `select_scatter`, then it is safe to assume the inputs are valid inputs to that function, otherwise we can let Torch throw the error.

Reply:
Ok got it. Thanks for the clarification!